src/common/ram/ram.go

```diff
@@ -43,6 +44,18 @@ func (res Resource) Subresource(resources ...Resource) Resource {
 	return Resource(path.Join(elements...))
 }
 
+// GetNamespace returns namespace from resource
+func (res Resource) GetNamespace() (Namespace, error) {
+	for _, parser := range namespaceParsers {
+		namespace, err := parser(res)
+		if err == nil {
+			return namespace, nil
+		}
+	}
+
+	return nil, errors.New("no namespace")
```

Comment:
Would be better to print the value of `resource` in the error message?

Reply:
resource info added